contract/src/api.rs

```diff
@@ -176,10 +184,25 @@ impl Contract {
             // First key receives all the deposit.
             attached_balance = 0;
         }
+
         if attached_balance > 0 {
             env::panic_str("The attached deposit could not be added to any account");
         }
-        SetReturnType { block_height: U64(near_sdk::env::block_height()) }
+
+        if options.callback_receiver_id.is_some() {
+            let callback_receiver_id = options.callback_receiver_id.unwrap();
+
+            callback_contract::on_social_set(
+                near_sdk::env::block_height(),
```

Comment:
For DevHub's purposes, to inform which account posted the discussion, we also need a list of updated SocialDB keys OR we should be able to pass free-form params through `set(options: { callback_params: {account_id} })`.

Getting the list of keys could be a bit of an involved process that would burn even more gas. Yet, free-form params can have security implications. Thus, I gravitate towards either passing the whole `data` or passing only the list of keys.